src/test/fuzz/addrman.cpp

```diff
@@ -114,4 +189,36 @@ FUZZ_TARGET_INIT(addrman, initialize_addrman)
     (void)addr_man.size();
     CDataStream data_stream(SER_NETWORK, PROTOCOL_VERSION);
     data_stream << addr_man;
+
+    // Check that serialize followed by unserialize produces the same addrman.
+
+    data_stream.clear();
+
+    const size_t num_sources = fuzzed_data_provider.ConsumeIntegralInRange<size_t>(1, 100);
+    for (size_t i = 0; i < num_sources; ++i) {
+        const size_t num_addresses = fuzzed_data_provider.ConsumeIntegralInRange<size_t>(1, 1000);
+        std::vector<CAddress> addresses(num_addresses);
+        for (CAddress& addr : addresses) {
+            const auto netaddr = ConsumeNetAddr(fuzzed_data_provider);
+            const auto port = fuzzed_data_provider.ConsumeIntegralInRange<uint16_t>(1, 65535);
+            const auto service = CService{netaddr, port};
+            const auto flags = ServiceFlags{fuzzed_data_provider.ConsumeIntegral<uint64_t>()};
+            addr = CAddress{service, flags};
+        }
+
+        const auto source = ConsumeNetAddr(fuzzed_data_provider);
+        const auto time_penalty = fuzzed_data_provider.ConsumeIntegralInRange<int64_t>(0, 100000000);
+        addr_man.Add(addresses, source, time_penalty);
+
+        // Move some to the "tried" table.
+        for (size_t j = 0; j < addresses.size(); j += 3) {
```

Comment:
seems this moves exactly 33%, but never 0% or 100%. Would be good allow the extreme cases as well.

If you feel fancy, you can write a templated helper `ChooseK(const size_t k_max, const Collection& collection, const UnaryOp& op)`, which picks at most `k_max` elements from the collection and applies `op` to them.

Reply:
Added the possibility that is moves 0% or 100% without adding `ChooseK()`.